data/transactions/logic/assembler.go

```diff
@@ -432,16 +432,88 @@ func parseBinaryArgs(args []string) (val []byte, consumed int, err error) {
 			return
 		}
 		consumed = 2
+	} else if len(arg) > 2 && arg[0] == '"' && arg[len(arg)-1] == '"' {
+		val, err = parseStringLiteral(arg)
+		consumed = 1
 	} else {
 		err = fmt.Errorf("byte arg did not parse: %v", arg)
 		return
 	}
 	return
 }
 
+func parseStringLiteral(input string) (result []byte, err error) {
+	start := 0
+	end := len(input) - 1
+	if input[start] != '"' || input[end] != '"' {
+		return nil, fmt.Errorf("no quotes")
+	}
+	start++
+
+	escapeSeq := false
+	hexSeq := false
+	result = make([]byte, 0, end-start+1)
+
+	// skip first and last quotes
+	pos := start
+	for pos < end {
+		char := input[pos]
+		if char == '\\' && !escapeSeq {
+			if hexSeq {
+				return nil, fmt.Errorf("escape seq inside hex number")
+			}
+			escapeSeq = true
+			pos++
+			continue
+		}
+		if escapeSeq {
+			escapeSeq = false
+			switch char {
+			case 'n':
+				char = '\n'
+			case 'r':
+				char = '\r'
+			case 't':
+				char = '\t'
+			case '\\':
+				char = '\\'
+			case '"':
+				char = '"'
+			case 'x':
+				hexSeq = true
+				pos++
+				continue
+			default:
+				return nil, fmt.Errorf("invalid escape seq \\%c", char)
+			}
+		}
+		if hexSeq {
+			hexSeq = false
+			if pos >= len(input)-2 { // count a closing quote
+				return nil, fmt.Errorf("non-terminated hex seq")
+			}
+			num, err := strconv.ParseUint(input[pos:pos+2], 16, 8)
+			if err != nil {
+				return nil, err
+			}
+			char = uint8(num)
+			pos++
```

Comment:
hm, should this be incrementing `pos` by 2?

Reply:
nope, it is a correcting +1 to feed one char and the second +1 for the second char is below.

Reply:
ah, I see now.